wire-ios-sync-engine/Source/UserSession/ZMUserSession.swift

```diff
@@ -550,6 +550,17 @@ public class ZMUserSession: NSObject {
         }
     }
 
+    private func observeOnEpochChangeIfNeeded() {
+        guard e2eiFeature.isEnabled,
+              let mlsService = self.syncContext.mlsService else {
+            return
+        }
+        let epochChangeToken = EpochChangeObserver(
+            mlsService: mlsService,
+            block: self.mlsConversationVerificationStatusProvider?.updateStatus(_:))
+        self.tokens.append(epochChangeToken)
+    }
```

Comment:
We should aim to not put logic inside the `ZMUserSession` this code could live in a `MLSConversationVerificationManager` class

Reply:
I removed EpochChangeObserver